runtime/js-compute-runtime/builtins/server-info.cpp

```diff
@@ -0,0 +1,101 @@
+#include "builtins/server-info.h"
+#include "core/geo_ip.h"
+#include "host_interface/host_api.h"
+#include "js/JSON.h"
+#include "openssl/evp.h"
+#include <arpa/inet.h>
+
+namespace builtins {
+
+namespace {
+
+JSString *address(JSObject *obj) {
+  JS::Value val = JS::GetReservedSlot(obj, static_cast<uint32_t>(ServerInfo::Slots::Address));
+  return val.isString() ? val.toString() : nullptr;
+}
+
+static JSString *retrieve_address(JSContext *cx, JS::HandleObject self) {
```

Comment:
Did you revert abstracting the IP bytes to JS string logic? I release we repeat the same routine for the request IP as well, does that mean we now have three functions doing the same thing? Perhaps reinvestigating sharing might be good.

Reply:
I abstracted it in the starling build but not in this build, I can abstract here also

Reply:
Ah sorry I missed which one I was looking at! Starling-only abstraction is fine, but specifically perhaps we can make it a shared util shared by the `Request.ip` one too.